File: src/main/java/com/lemonzuo/license/common/service/RpcService.java
package com.lemonzuo.license.common.service;

/**
 * @author LemonZuo
 * @create 2024-05-16 23:33
 */
public interface RpcService {
    /**
     * ping
     * @param machineId 机器码
     * @param salt 盐
     * @return pong
     */
    String ping(String machineId, String salt);


    /**
     * 获取授权
     * @param username 用户名
     * @param hostName 主机名
     * @param machineId 机器码
     * @param salt 盐
     * @return 授权信息
     */
    String obtainTicket(String username, String hostName, String machineId, String salt);

    /**
     * 续期
     * @param machineId 机器码
     * @param salt 盐
     * @return 续期信息
     */
    String releaseTicket(String machineId, String salt);
}

File: rpc/service/rpcService.go
package service

import "github.com/gin-gonic/gin"

// RpcService 定义了与远程过程调用相关的方法
type RpcService interface {
	Ping(ctx *gin.Context, machineId, salt string)
	ObtainTicket(ctx *gin.Context, username, hostName, machineId, salt string)
	ReleaseTicket(ctx *gin.Context, machineId, salt string)
}
